### PR TITLE
Improvements to your paper

### PR DESCRIPTION
**Pull request by fu**
Hi Guys,

I though I use the power of github on your github paper, Yasset encouraged me to create a pull request, thus here we go :) In the beginning I just wanted to add the auto documentation feature that we are using most of the time for our projects, but I got taken away. Especially the old Rule 6&7 were sort of the same, Rule 8&9 had not much to do with github, or at least one had to adjust the text a bit and I had to delete Rule 10. I know it is a lot, but maybe you like it. 

Best regards,

.c

### CODE REVIEW

**Comment by lgatto** [comment]
Dear @fu 
Thank you for your contribution.

Would you mind summarising the major changes you did here (rules 6/7 and 8/9), so that we can discuss them. That will make it easier for everybody to chime in rather than one persons fixing all the conflicts - there have been additional (relatively minor) updates since your fork and update.

**Comment by ypriverol** [comment]
@fu Agree with @lgatto the manuscript. Pelase make a summary of the major changes and keep in mind that the manuscript is almost ready to be submitted in my opinion if they are major changes we can include then after the first review iteration.

**Comment by lgatto** [comment]
@ypriverol if we consider major changes, we should do them before submission.

**Comment by ypriverol** [comment]
Yes I know, but only if these changes do not generate a huge loop and discussions because it can delay more the submission. In any case in my previous talks with @fu I understand that were minor changes. Please @fu make clear everything as soon as possible to move forward the project.

**Comment by fu** [comment]
Hi @ypriverol and @lgatto,

of course. Sorry that I just hacked stuff in without leaving much of a trail. So here we go
In the abstract I only did minor changes

Rule1 I added the forking progress since it was missing and the fact that while workin on a fork one still can incorporate the ongoing developments of the master branch is IMHO an important point. I also added a statement about big file which should not be part of the git if not needed (if you do automatised testing and so on ... Rule 6)

Rule 2 "Learn Git and embrace its power" was splitted and I created an additional new rule 3 = Use branches (pretty much the same text)

Old Rule 3 "Let others contribute and add ideas to your projects" was renamed to Rule 7. "Use and maintain your github issue trackers" - I though the flow of the rules should somehow be connected with the dev process ... learn git - use branches - label and tag - follow best practises - work on issues - then the additional stuff. 

Rule 4 is now new. Introduced the Tag feature and the explanation why they (and semantic versioning) are important - also in the light of the "cite source code block"

Old Rule 4 "The code must always be ready to use: Make use of tests, deployments and continuous integration " was splitted into your major part "Follow basic principles for continuous integration" (now rule 4) and the post-processing hocks that enable the programmer to "Follow basic principles for continuous integration" -> now rule 5 : "Use of the power of post-processing steps on github to show that your produce high quality code"

New Rule 5 "Use of the power of post-processing steps on github to show that your produce high quality code" was extended by auto documentation feature which was a important thing missing (IMHO). So the paper would recommend to a) use travis , b) codecov and c) autodoc generation. Additionally I introduced badges.

Old Rule 5 and Rule 8 are sort of the same - enhancing visibility of your github project using github. I merge these to Rule 9 "Promote your project/s in the scientific community - create a web page and more"

Rule 6 and 7 also are in the same ballpark and were merged into "Rule 8. Make your code easily citable, and cite source code!" 

Rule 9 became Rule 10

and Rule 10 was removed- IMHO I don't think this is important enough to create a point (could go into new 10 or Rule 8. 

I know that is a lot and really no hard feelings, if you want to skip my comments for your submission. That's what git is all about :smile:

**Comment by lgatto** [comment]
My comments, while I read both versions and before I adapt and merge
- Intro: minor modifications, which I think generally improve the text. 
- Rule 1: forking is a good addition and flows really well here. Will slightly rephrase the fragmentation bit. Not sure if we need to go in many details about large files (re the `--maximum-depth`) but might be useful to add this work of caution. Will merge and then discuss with @javizca when we do the final proof-reading before submission. 
- Rule 2/3: I think it is worth have a rule for branches - they are so easy to use in git. 
- Rule 4: why not - will have to re-evaluate in the light of the whole thing. At this stage, we could actually also mention assignment of DOIs to specific versions, which is mentioned later, as far as I remember.
- Rule 5 (was rule 4): don't like new title, as it does not mean much for somebody that is new to the concepts. Not clear why current rule 5 and 6 should we separated. I think the difference between CI and hooks is not really relevant here - the main message is automation. I think it's fine to mention badges (also sometimes called shields), although maybe not essential. 
- I think the rest is essentially stuff form former rules moved. 
- Rule 10: we need a 10th rule! We could keep and improve what we have but explicitly mentioning following github users and watching issues. 

In summary, I will merge and fix conflicts into the latest version. @fu's version is not dramatically different from what we have, just re-structured. I would vote for the new rules order, if only because the order was motivated 

> I though the flow of the rules should somehow be connected with the dev process ... learn git - use branches - label and tag - follow best practises - work on issues - then the additional stuff.

and now seems more logical. I think we should motivate this in the intro; this will help readers, I think.
